app/src/main/java/com/willowtree/vocable/MainActivity.kt

```diff
@@ -1,52 +1,36 @@
 package com.willowtree.vocable
 
-import android.Manifest
-import android.app.ActivityManager
 import android.content.Context
 import android.graphics.Rect
-import android.hardware.display.DisplayManager
 import android.os.Bundle
-import android.util.DisplayMetrics
-import android.view.Surface
 import android.view.View
 import android.view.ViewGroup
-import android.widget.Toast
-import androidx.appcompat.app.AppCompatActivity
 import androidx.core.view.children
 import androidx.core.view.isVisible
-import com.google.ar.core.ArCoreApk
-import com.vmadalin.easypermissions.EasyPermissions
-import com.vmadalin.easypermissions.dialogs.SettingsDialog
 import com.willowtree.vocable.customviews.PointerListener
 import com.willowtree.vocable.customviews.PointerView
 import com.willowtree.vocable.databinding.ActivityMainBinding
-import com.willowtree.vocable.facetracking.FaceTrackFragment
 import com.willowtree.vocable.facetracking.FaceTrackingViewModel
-import com.willowtree.vocable.settings.selectionmode.HeadTrackingPermissionState
-import com.willowtree.vocable.settings.selectionmode.SelectionModeViewModel
-import com.willowtree.vocable.utils.VocableSharedPreferences
+import com.willowtree.vocable.utils.FaceTrackingManager
+import com.willowtree.vocable.utils.FaceTrackingPointerUpdates
+import com.willowtree.vocable.utils.IVocableSharedPreferences
 import com.willowtree.vocable.utils.VocableTextToSpeech
 import io.github.inflationx.viewpump.ViewPumpContextWrapper
 import org.koin.android.ext.android.inject
+import org.koin.androidx.scope.ScopeActivity
 import org.koin.androidx.viewmodel.ViewModelOwner
 import org.koin.androidx.viewmodel.ext.android.viewModel
-import timber.log.Timber
 
-class MainActivity : AppCompatActivity(),
-                     EasyPermissions.PermissionCallbacks {
+class MainActivity : ScopeActivity() {
 
-    private val minOpenGlVersion = 3.0
-    private val displayMetrics = DisplayMetrics()
     private var currentView: View? = null
     private var paused = false
     private lateinit var binding: ActivityMainBinding
-    private val sharedPrefs: VocableSharedPreferences by inject()
-    private var hasSetupAr: Boolean = false
+    private val sharedPrefs: IVocableSharedPreferences by inject()
     private val allViews = mutableListOf<View>()
 
-    private val selectionModeViewModel: SelectionModeViewModel by viewModel(owner = {
-        ViewModelOwner.from(this)
-    })
+    private val faceTrackingManager: FaceTrackingManager by inject()
```

Comment:
The `SelectionModeViewModel` has the `FaceTrackingPermissions` object, not the manager. But since the manager deals with `Activity` and `View` objects it feels sort of weird to put through a ViewModel?